src/main/java/com/wavefront/internal/SpanDerivedMetricsUtils.java

```diff
@@ -0,0 +1,207 @@
+      pointTags.put(SERVICE_TAG_KEY, getNonEmptyOrDefaultValue(service, "unknown_service"));
+      pointTags.put(CLUSTER_TAG_KEY, getNonEmptyOrDefaultValue(cluster, NULL_TAG_VAL));
+      pointTags.put(SHARD_TAG_KEY, getNonEmptyOrDefaultValue(shard, NULL_TAG_VAL));
+      pointTags.put(OPERATION_NAME_TAG, getNonEmptyOrDefaultValue(operationName, "unknown_operation"));
+      pointTags.put(COMPONENT_TAG_KEY, getNonEmptyOrDefaultValue(componentTagValue, NULL_TAG_VAL));
+      pointTags.put(SOURCE_KEY, source);
+
+      if (traceDerivedCustomTagKeys != null && traceDerivedCustomTagKeys.size() > 0) {
+        spanTags.forEach((tag) -> {
+          String tagKey = tag._1;
+          String tagValue = tag._2;
+          if (traceDerivedCustomTagKeys.contains(tagKey)) {
+            pointTags.put(tagKey, tagValue);
+          }
+          // propagate http status
+          if (tagKey.equalsIgnoreCase(HTTP_STATUS_KEY)) {
+            pointTags.put(HTTP_STATUS_KEY, tagValue);
+          }
+        });
+      }
+
+      // span.kind tag will be promoted by default
+      pointTags.putIfAbsent(SPAN_KIND_KEY, NULL_TAG_VAL);
+
+      // tracing.derived.<application>.<service>.<operation>.invocation.count
+      wfInternalReporter.newDeltaCounter(new MetricName(sanitizeWithoutQuotes(application +
+          "." + service + "." + operationName + INVOCATION_SUFFIX), pointTags)).inc();
+
+      if (isError) {
+        // tracing.derived.<application>.<service>.<operation>.error.count
+        wfInternalReporter.newDeltaCounter(new MetricName(sanitizeWithoutQuotes(application +
+            "." + service + "." + operationName + ERROR_SUFFIX), pointTags)).inc();
+      }
+
+      // tracing.derived.<application>.<service>.<operation>.duration.micros.m
+      if (isError) {
+        Map<String, String> errorPointTags = new HashMap<>(pointTags);
+        errorPointTags.put("error", "true");
+        wfInternalReporter.newWavefrontHistogram(new MetricName(sanitizeWithoutQuotes(application +
+            "." + service + "." + operationName + DURATION_SUFFIX), errorPointTags), clock).
+            update(spanDurationMicros);
+      } else {
+        wfInternalReporter.newWavefrontHistogram(new MetricName(sanitizeWithoutQuotes(application +
+            "." + service + "." + operationName + DURATION_SUFFIX), pointTags), clock).
+            update(spanDurationMicros);
+      }
+
+      // tracing.derived.<application>.<service>.<operation>.total_time.millis.count
+      wfInternalReporter.newDeltaCounter(new MetricName(sanitizeWithoutQuotes(application +
+          "." + service + "." + operationName + TOTAL_TIME_SUFFIX), pointTags)).
+          inc(spanDurationMicros / 1000);
+
+      // Remove operation tag and source tag from tags list before sending RED heartbeat.
+      pointTags.remove(OPERATION_NAME_TAG);
+      pointTags.remove(SOURCE_KEY);
+    } catch (Exception ignored) {
+      // This should never happen, if all SDKs version are set properly.
```

Comment:
why is the try-catch needed? also, should `pointTags.remove()` be outside of the try-catch in case there is an exception?

Reply:
The try-catch here is because I notice when the maven is configured with incorrect SDKs version (happens to me when I try to update sample-app-java to test this PR), you'll get the exception of `RuntimeException: tried to get nonexistent method sanitizeWithoutQuotes`, which blocked the dropwizard application to start. So I feel it might be safer to try-catch here.

Reply:
and yes, `pointTags.remove()` should be outside of the try-catch

Reply:
I dont think we should put this in try and catch and ignore the exception.
swallowing the exception is not a right idea here.
if we really care about try /catch we should fail fast and bubble up the exception to the caller.
the sdk mismatch happened because we did not use the right version of the sdks. I think that's a user-error.

Reply:
try-catch block removed